src/main.rs

```diff
@@ -1,3 +1,132 @@
+#[macro_use]
+extern crate clap;
+
+mod options;
+use crate::options::CLIOptions;
+use clap::App;
+use std::{fs, fs::File, io, path::Path};
+
 fn main() {
-    println!("Hello, world!!!");
+    let yaml = load_yaml!("../cli_def/en_us.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    let raw_options = CLIOptions::from_clap_matches(&matches);
+
+    let options = raw_options.process_options();
+
+    for filename in options.files {
+        let file = File::open(filename).unwrap();
+        unzip_file(
+            file,
+            &std::path::PathBuf::from(&raw_options.destination_folder),
+        );
+    }
+}
+
+fn unzip_file(file: File, directory: &Path) {
+    let mut archive = zip::ZipArchive::new(file).unwrap();
+
+    let files_indexes = 0..archive.len();
+
+    files_indexes
+        .map(|file_index| unzip_archive(file_index, &mut archive, directory))
+        .for_each(drop);
```

Comment:
Not entirely sure why you need this.

Reply:
Yeah, that remains from when I was trying to extract the zip in parallel. I used iterators for trying to use rayon. But it didn't work because the library is not build in a ways that allow multi-threading. I will change it to a for lop.